redisson/src/main/java/org/redisson/connection/balancer/LoadBalancerManager.java

```diff
@@ -148,9 +149,18 @@ protected void onSuccess(Void value) {
                         }
                     };
                     listener.setCounter(2);
-                    BiConsumer<Void, Throwable> initCallBack = (r, ex) -> {
-                        if (ex == null) {
-                            listener.decCounter();
+                    BiConsumer<Void, Throwable> initCallBack = new BiConsumer<Void, Throwable>() {
+                        private AtomicBoolean initConnError = new AtomicBoolean(false);
+                        @Override
+                        public void accept(Void r, Throwable ex) {
+                            if (ex == null) {
+                                listener.decCounter();
+                            } else {
+                                if (!initConnError.compareAndSet(false, true)) {
```

Comment:
do we still need this check?

Reply:
i think we need this check, beacuse initCallBack will called by slaveConnectionPool and pubSubConnectionPool completed, but only can reset initialized status once